client/my-sites/comments/comment/comment-edit.jsx

```diff
@@ -41,20 +46,46 @@ export class CommentEdit extends Component {
 		authorDisplayName: '',
 		authorUrl: '',
 		commentContent: '',
+		commentDate: '',
+		isDatePopoverVisible: false,
+		tmpCommentDate: '',
 	};
 
 	componentWillMount() {
-		const { authorDisplayName, authorUrl, commentContent } = this.props;
-		this.setState( { authorDisplayName, authorUrl, commentContent } );
+		const { authorDisplayName, authorUrl, commentContent, commentDate } = this.props;
+		this.setState( { authorDisplayName, authorUrl, commentContent, commentDate } );
 	}
 
+	storeDatePopoverButtonRef = button => ( this.datePopoverButtonRef = button );
+
+	toggleDatePopover = () =>
+		this.setState( ( { commentDate, isDatePopoverVisible } ) => ( {
+			isDatePopoverVisible: ! isDatePopoverVisible,
+			tmpCommentDate: isDatePopoverVisible ? '' : commentDate,
+		} ) );
+
+	cancelCommentDataValueChange = () =>
+		this.setState( ( { tmpCommentDate } ) => ( {
+			commentDate: tmpCommentDate,
+			isDatePopoverVisible: false,
+			tmpCommentDate: '',
+		} ) );
+
+	getTimezoneForPostSchedule = () => ( {
+		timezone: this.props.siteTimezone || undefined,
+		gmtOffset: parseInt( this.props.siteGmtOffset, 10 ),
+	} );
+
 	setAuthorDisplayNameValue = event => this.setState( { authorDisplayName: event.target.value } );
 
 	setAuthorUrlValue = event => this.setState( { authorUrl: event.target.value } );
 
 	setCommentContentValue = ( event, callback = noop ) =>
 		this.setState( { commentContent: event.target.value }, callback );
 
+	setCommentDateValue = commentDate =>
+		this.setState( { commentDate: this.props.moment( commentDate ).format() } );
```

Comment:
would it make sense to store a `Date` object here vs. a time string?

Reply:
`state.commentDate` is what will be sent to the update endpoint, which expects an ISO 8601 datetime (e.g. `2018-06-20T14:48:15+02:00`), which is what `moment().format()` outputs.

Reply:
nice answer to a different question 😉 

of course we want to send a serialized time stamp when issuing API calls

does it make sense to store that _here_ though? `Date` is _the_ native Javascript format for representing dates and times and timezones. we don't have to make this change, but just remember that we aren't powerless to choose internal representations either 👍

Reply:
But then we'd basically end up adding a new conversion, or moving the burden, somewhere else.
In this particular case, we don't really need to do anything with this `Date`, and all the components and logic involved expect a string.

Reply:
if you can make a case for it that's fine. the burden is probably going to be high in general though to store date values internally as a string. we end up with thirty date values and each one is stored differently…